app/Http/Controllers/VendorController.php

```diff
@@ -2,110 +2,55 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\User;
 use Illuminate\Http\Request;
+use Illuminate\Validation\Rules;
+use Illuminate\Http\UploadedFile;
 use Illuminate\Support\Facades\Auth;
-use Illuminate\Http\RedirectResponse;
 use Illuminate\Support\Facades\Hash;
-use Illuminate\Validation\Rules;
-use App\Providers\RouteServiceProvider;
+use Illuminate\Http\RedirectResponse;
 use Illuminate\Auth\Events\Registered;
-use App\Models\User;
 use App\Notifications\AccountStatusChanged;
-use App\Http\Middleware\VendorStatusChecker;
+use App\Http\Requests\Vendor\VendorDataRequest;
+use App\Services\Notification\NotificationService;
 
 class VendorController extends Controller
 {
-    // Dashboard view
-    public function vendorDashboard() 
+    // Dashboard view ** REFACTORED **
+    public function index() 
     {
         return view('vendor.index');
-
     }
 
     // Vendor Login
     public function vendorLogin()
     {   
-        // The middleware has already checked the status
         return view('vendor.vendor_login');
     }
 
-    // Vendor logout
-    public function vendorLogout(Request $request): RedirectResponse 
+    // Vendor logout ** REFACTORED **
+    public function logout(NotificationService $notification): RedirectResponse 
     {
-        //Logout the user
         Auth::guard('web')->logout();
+        session()->invalidate();
+        session()->regenerateToken();
 
-        //Invalidate the session
-        $request->session()->invalidate();
-
-        //Regenerate the CSRF token
-        $request->session()->regenerateToken();
-
-        //Redirect to the vendor login page
-        return redirect('/vendor/login');
+        return redirect('/vendor/login')->with($notification->message('You have been logged out successfully', 'success'));
```

Comment:
You could use the route names when redirecting as in `redirect()->route('my.route');`